src/replica/split/replica_split_manager.cpp

```diff
@@ -1434,5 +1444,65 @@ void replica_split_manager::parent_send_notify_stop_request(
     });
 }
 
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_split_manager::query_child_state() // on primary parent
+{
+    auto request = make_unique<query_child_state_request>();
+    request->app_name = _replica->_app_info.app_name;
+    request->pid = get_gpid();
+    request->partition_count = _replica->_app_info.partition_count;
+
+    rpc_address meta_address(_stub->_failure_detector->get_servers());
+    ddebug_replica("send query child partition state request to meta server({})",
+                   meta_address.to_string());
+    query_child_state_rpc rpc(
+        std::move(request), RPC_CM_QUERY_CHILD_STATE, 0_ms, 0, get_gpid().thread_hash());
+    _replica->_primary_states.query_child_task =
+        rpc.call(meta_address, tracker(), [this, rpc](error_code ec) mutable {
+            on_query_child_state_reply(ec, rpc.request(), rpc.response());
+        });
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_split_manager::on_query_child_state_reply(
+    error_code ec,
+    const query_child_state_request &request,
+    const query_child_state_response &response) // on primary parent
+{
+    _replica->_checker.only_one_thread_access();
+
+    if (ec != ERR_OK) {
+        dwarn_replica("query child partition state failed, error = {}, retry it later", ec);
+        _replica->_primary_states.query_child_task =
+            tasking::enqueue(LPC_PARTITION_SPLIT,
+                             tracker(),
+                             std::bind(&replica_split_manager::query_child_state, this),
+                             get_gpid().thread_hash(),
+                             std::chrono::seconds(1));
+        return;
+    }
+
+    if (response.err != ERR_OK) {
+        dwarn_replica("app({}) partition({}) split has been canceled, ignore it",
+                      request.app_name,
+                      request.pid);
+        return;
+    }
+
+    ddebug_replica("query child partition succeed, child partition[{}] has already been ready",
+                   response.child_config.pid);
+    // make child partition active
+    _stub->split_replica_exec(LPC_PARTITION_SPLIT,
+                              response.child_config.pid,
+                              std::bind(&replica_split_manager::child_partition_active,
+                                        std::placeholders::_1,
+                                        response.child_config));
+    update_local_partition_count(response.partition_count);
+    _replica->_primary_states.cleanup_split_states();
```

Comment:
What about move `_replica->_primary_states.cleanup_split_states();` into `parent_cleanup_split_context();`

Reply:
It is a good recommend~ Actually, I will refactor it after I merge all function pull requests.